ADD option.minLength

pubkey opened the pull request:
This PR adds 2 parameters to the options.

1. `minLength`
Words with a character-amount of less than `minLength` will not be spell-checked.
This is usefull for abbreviations which mostly have only up to 3 characters and are currently marked as typo when running the lint. This requires the user to add many words to the ignore-list by hand.
The default-value of minLength is `1` which means no word is skipped by this PR.

2. `skipWordIfMatch`
Words that match one of the given regex-expressions will not be spell-checked.
This is usefull when you have prefixes on variables. It also occurs that many issues are created because people thinkg that `skipIfMatch` does exactly this. (like [#29](https://github.com/aotaduy/eslint-plugin-spellcheck/issues/29) and [#30](https://github.com/aotaduy/eslint-plugin-spellcheck/issues/30))

aotaduy commented on README.md:
```diff
@@ -29,6 +29,9 @@ Array of words that will not be checked.
 i.e: "^[-\\w]+\/[-\\w\\.]+$" will ignore MIME types.
+
+"minLength": <<Number>> default: 1
+Words with a character-amount of less than the minLength will not be spell-checked.
```
Please change to:
js node element value (identifier, comment, string, string template, etc) with a character-amount of less than the minLength will not be spell-checked.

Because we also divide the token for example for camelCalse or snakeCase and then check the spelling inside each word. And this option applies only to the complete node content.

pubkey replied:
Looks like this PR does not exactly what I wanted to do.
I will do some changes so the code does what the description tells.

aotaduy replied:
Ok cool no problem, I think that you want to filter any subsring with less than x chars right? 
Not the entire node. So if a variable is idElement you don't want to spell check id. 
That is ok for me too.

pubkey replied:
Yes that is the goal. PR will come soon.

pubkey commented:
I changed some things and also added `skipWordIfMatch`.
Both `minLength` and `skipWordIfMatch` will now skip the spell-checking of single words instead of the entire nodes.
I updated the initial PR-text.

pubkey commented:
Is there a problem with this PR?
Shell I split it into 2 PRs one with minLength and one with skipWordIfMatch?

aotaduy commented:
No problem at all. 
Didn't have the time to review.

pubkey commented:
Any update on this?
